### PR TITLE
Updated version

### PR DESCRIPTION
**Pull request by nickImbirev**
Signed-off-by: Nikolai Imbirev <nikolay.imbirev@broadcom.com>

### CODE REVIEW

**Comment by jellypuno** [comment]
@nickImbirev don't forget to generate the license text in the files and also the version (v0.12.0)

**Comment by IgorCATech** [comment]
Is there any doc/text that I should look at?

**Comment by nickImbirev** [comment]
> Is there any doc/text that I should look at?

Nope, this is just a copy-paste stuff, so there was no changes since your last updates

**Comment by nickImbirev** [comment]
> @nickImbirev don't forget to generate the license text in the files and also the version (v0.12.0)

Done, thank you, good catch!

File: package.json [hunk 1]
```diff
@@ -1,337 +1,71 @@
...
-    "@broadcom/endevor-for-zowe-cli": "^5.7.2",
-    "@zowe/imperative": "^4.7.4"
+    "typescript": "^4.1.3",
+    "vscode-test": "^1.4.1",
+    "webpack": "^4.44.2",
+    "webpack-cli": "^4.1.0",
+    "@local/eslint-config-local": "1.0.0",
+    "@types/selenium-webdriver": "^3.0.4",
+    "selenium-webdriver": "^3.6.0",
+    "testcontainers": "^7.4.0"
```
Comment (deepali-hub) [inline package.json:69]: do we need `testcontainers` dependency?

**Comment by katelynienaber** [comment]
Hello, I am getting this error when I pull from your fork:
image

**Comment by nickImbirev** [comment]
Soo, what we have right now in our pipeline:
1. We can run this using `yarn` commands: `lint`, `test`, `pretty`, `build`, `package`.
2. Initial package.json is adjusted for old Jenkinsfile (so it is not recommended to use commands from there)
3. Our package process is a dummy script - so we need to disable deploy in development (without it - we will deploy incorrect vsix file into Che downloads page).

I can propose to fix Jenkinsfile after merging this PR immediately and disable deploy for now.

File: package.json (hunk 1 above)
Comment (nickImbirev) [reply package.json]: Yep, we don't need it now. I deleted it, thank you!

**Comment by nickImbirev** [comment]
> Hello, I am getting this error when I pull from your fork:
> image
> 
> I made extra sure I have the current code.

I think, there is a git problem here, related to deleted `src` folder. It should be out of the git index now, and it will be deleted in the repo itself. I can propose to delete it locally too and the problem will gone. 
Did it help?

**Comment by nickImbirev** [comment]
This PR depends on this one: https://github.com/eclipse/che-che4z-explorer-for-endevor/pull/99

**Comment by katelynienaber** [comment]
> Soo, what we have right now in our pipeline:
> 
> 1. We can run this code using `yarn` commands: `lint`, `test`, `pretty`, `build`, `package`.
> 2. Root package.json is adjusted for old Jenkinsfile (so it is not recommended to use commands from there)
> 3. Our package process is a dummy script - so we need to disable deploy in development (without it - we will deploy incorrect vsix file into Che downloads page).
> 
> I can propose to fix Jenkinsfile after merging this PR immediately and disable deploy for now.

I can't run `yarn build`, command not found, but lint works.

**Comment by katelynienaber** [comment]
> > Hello, I am getting this error when I pull from your fork:
> > image
> > I made extra sure I have the current code.
> 
> I think, there is a git problem here, related to deleted `src` folder. It should be out of the git index now, and it will be deleted in the repo itself. I can propose to delete it locally too and the problem will gone.
> Did it help?

Where do i go to remove this?

**Comment by nickImbirev** [comment]
> > Soo, what we have right now in our pipeline:
> > 
> > 1. We can run this code using `yarn` commands: `lint`, `test`, `pretty`, `build`, `package`.
> > 2. Root package.json is adjusted for old Jenkinsfile (so it is not recommended to use commands from there)
> > 3. Our package process is a dummy script - so we need to disable deploy in development (without it - we will deploy incorrect vsix file into Che downloads page).
> > 
> > I can propose to fix Jenkinsfile after merging this PR immediately and disable deploy for now.
> 
> I can't run `yarn build`, command not found, but lint works.

Hmm, are you sure, that you run in the repo root? Did you ran `yarn` before to refresh dependencies?

**Comment by nickImbirev** [comment]
> > > Hello, I am getting this error when I pull from your fork:
> > > image
> > > I made extra sure I have the current code.
> > 
> > 
> > I think, there is a git problem here, related to deleted `src` folder. It should be out of the git index now, and it will be deleted in the repo itself. I can propose to delete it locally too and the problem will gone.
> > Did it help?
> 
> Where do i go to remove this?

Locally, in your local repo version. 

The problem is: git is saving unindexed folders, even if they were deleted in PR. So you have to remove them manually.

**Comment by katelynienaber** [comment]
> > > Soo, what we have right now in our pipeline:
> > > 
> > > 1. We can run this code using `yarn` commands: `lint`, `test`, `pretty`, `build`, `package`.
> > > 2. Root package.json is adjusted for old Jenkinsfile (so it is not recommended to use commands from there)
> > > 3. Our package process is a dummy script - so we need to disable deploy in development (without it - we will deploy incorrect vsix file into Che downloads page).
> > > 
> > > I can propose to fix Jenkinsfile after merging this PR immediately and disable deploy for now.
> > 
> > 
> > I can't run `yarn build`, command not found, but lint works.
> 
> Hmm, are you sure, that you run in the repo root? Did you run `yarn` before to refresh dependencies?

Yes I am in root, and i get same response if i run it after yarn command:

image

**Comment by katelynienaber** [comment]
> > > > Hello, I am getting this error when I pull from your fork:
> > > > image
> > > > I made extra sure I have the current code.
> > > 
> > > 
> > > I think, there is a git problem here, related to deleted `src` folder. It should be out of the git index now, and it will be deleted in the repo itself. I can propose to delete it locally too and the problem will gone.
> > > Did it help?
> > 
> > 
> > Where do i go to remove this?
> 
> Locally, in your local repo version.
> 
> The problem is: git is saving unindexed folders, even if they were deleted in PR. So you have to remove them manually.

I don't understand. You want me to delete the src folder with all the code? How will it run without the code?

**Comment by nickImbirev** [comment]
> > > > Soo, what we have right now in our pipeline:
> > > > 
> > > > 1. We can run this code using `yarn` commands: `lint`, `test`, `pretty`, `build`, `package`.
> > > > 2. Root package.json is adjusted for old Jenkinsfile (so it is not recommended to use commands from there)
> > > > 3. Our package process is a dummy script - so we need to disable deploy in development (without it - we will deploy incorrect vsix file into Che downloads page).
> > > > 
> > > > I can propose to fix Jenkinsfile after merging this PR immediately and disable deploy for now.
> > > 
> > > 
> > > I can't run `yarn build`, command not found, but lint works.
> > 
> > 
> > Hmm, are you sure, that you run in the repo root? Did you run `yarn` before to refresh dependencies?
> 
> Yes I am in root, and i get same response if i run it after yarn command:
> 
> image

Are you sure, that you are in `e4e-dump` branch?

**Comment by nickImbirev** [comment]
> > > > > Hello, I am getting this error when I pull from your fork:
> > > > > image
> > > > > I made extra sure I have the current code.
> > > > 
> > > > 
> > > > I think, there is a git problem here, related to deleted `src` folder. It should be out of the git index now, and it will be deleted in the repo itself. I can propose to delete it locally too and the problem will gone.
> > > > Did it help?
> > > 
> > > 
> > > Where do i go to remove this?
> > 
> > 
> > Locally, in your local repo version.
> > The problem is: git is saving unindexed folders, even if they were deleted in PR. So you have to remove them manually.
> 
> I don't understand. You want me to delete the src folder with all the code? How will it run without the code?

Hmm, the code is moved to the `packages` folder. So we don't need to have `src` folder in the repo root. Maybe we are watching on different branches?

**Comment by katelynienaber** [comment]
I tested the following:
- Generate
- Edit
- Retrieve
- Retrieve with deps
- View details

Very cool thing, while retrieving a giant list of elements, you can still do other stuff like View details, create new profile, etc.

One thing, not a huge deal though. I got this error:
image
By doing the following:
- Create bogus Endevor profile
- Add a location profile to it
- Try to expand the location profile (so now getChildren has been called)
- While the extension is trying to load children...
- Try to remove the Endevor profile - this is successful, but profile is still in tree at this point
- Try to remove Endevor profile again
- Error appears
I think the only problem here is that a weird error shows up for the user, maybe we can silence it:
image

Another thing (maybe doens't need to be dealt with now)...when you retrieve lots of elements, the info dialog gets bigger:
image

Maybe we could limit the number of elements shown here to 5, and say, "see full list in output" or something like that?

And last but not least, the validation errors when creating a new location profile are still unclear, maybe we can fix this real quick...idk also I can do it but I'd want to make it really consider every portion of the path as one field and add validation there and update the message, maybe it's too much to add fast.
image

**Review by jellypuno — APPROVED** [review]
I was able to do a quick test of the functions and it is working fine with me.

**Comment by jellypuno** [comment]
@katelynienaber If you are good with this PR, kindly approve it. After that @zdmullen can merge it.

**Pull request merged**